index.js

```diff
@@ -4,21 +4,27 @@ var http = require('http');
 var https = require('https');
 var util = require('util');
 
-function getProtocolURI(end_point) {
+function getProtocolURIPort(end_point) {
   var mapping = {};
+  var uri_port = [];
   if (end_point.toLowerCase().indexOf("http://") === 0) {
```

Comment:
I'm nervous about this parsing.  

Let's use Node's library:
https://nodejs.org/api/url.html#url_url